README.md

```diff
@@ -17,19 +17,49 @@ conda activate c3po-os-api
 
 7. Install dependencies and dev dependencies locally from the `Pipfile` by running `pipenv install --dev`.
 
-8. Start the app `pipenv run uvicorn app.api:versioned_app`.
-
-9. Post a request in a separate shell!
-```bash
-curl -X POST -H "Content-Type: application/json" \
-    -d '{"text": "A fun little note."}' \
-    http://localhost:8000/predict
+8.  Create a `secrets.env` file and ensure it is on the `.gitignore`.  Add the following for local dev:
+```sh
+MINIO_ACCESS_KEY="<from console>"
+MINIO_SECRET_KEY="<from console>"
+MINIO_ENDPOINT_URL="http://localhost:9000"
+
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=postgres
+POSTGRES_SERVER=localhost
+POSTGRES_PORT=5432
+POSTGRES_DB=postgres
+
+MM_TOKEN="<your_preprod_mattermost_token>"
 ```
 
-10. Run tests and get coverage with `pytest --cov=app`
+9. Launch postgres, pgadmin, and minio via docker-compose `docker-compose up --build`.
+
+10. Visit `localhost:9001`.  Login with user:`miniouser` and password:`minioadmin`.  This is the minio console.
+
+11. Visit `localhost:5050`.  Login with email:`user@test.com` and password:`admin`.  This is the pgadmin console.  **See notes below for important details**
+
+12. Run the app db init script `./scripts/prestart.sh`
+
+13. Keeping your docker containers running, start the app in a new terminal (activate your conda env first) with `pipenv run uvicorn app.main:versioned_app --reload`.
+
+14. Open `localhost:8000/v1/docs` and start interacting with swagger!
+
+15. Run tests and get coverage with `pytest --cov=app`
+
+16.  You can shut down and your db / minio data will persist via docker volumes.
+
 
 # Notes
```

Comment:
having trouble configuring pgadmin to see the tables were created properly

Reply:
discuss today